torch/nn/parallel/distributed.py

```diff
@@ -1187,7 +1189,8 @@ def register_comm_hook(self, state: object, hook: callable):
             Gradbucket tensors should consist of only a single tensor.
 
         .. warning ::
-            ``get_future`` API supports only NCCL backend and will return a ``torch._C.Future``
+            ``get_future`` API supports NCCL, and partially GLOO and MPI backends (support is limited
```

Comment:
Does it mean that DDP comm hook (assume it only invokes allreduce API and no send/recv) can be supported on all the 3 backends now?

Reply:
yes